libsyclinterface/source/dpctl_sycl_device_interface.cpp

```diff
@@ -903,3 +902,117 @@ DPCTLDevice_GetCompositeDevice(__dpctl_keep const DPCTLSyclDeviceRef DRef)
     else
         return nullptr;
 }
+
+bool DPCTLDevice_CanAccessPeer(__dpctl_keep const DPCTLSyclDeviceRef DRef,
+                               __dpctl_keep const DPCTLSyclDeviceRef PDRef,
+                               DPCTLPeerAccessType PT)
+{
+    bool canAccess = false;
+    auto D = unwrap<device>(DRef);
+    auto PD = unwrap<device>(PDRef);
+    if (D && PD) {
+        auto BE1 = D->get_backend();
+        auto BE2 = PD->get_backend();
+
+        if (BE1 != sycl::backend::ext_oneapi_level_zero &&
+            BE1 != sycl::backend::ext_oneapi_cuda &&
+            BE1 != sycl::backend::ext_oneapi_hip)
+        {
+            error_handler("Backend " + std::to_string(static_cast<int>(BE1)) +
+                              " does not support peer access",
+                          __FILE__, __func__, __LINE__);
+            return false;
+        }
+
+        if (BE2 != sycl::backend::ext_oneapi_level_zero &&
+            BE2 != sycl::backend::ext_oneapi_cuda &&
+            BE2 != sycl::backend::ext_oneapi_hip)
+        {
+            error_handler("Backend " + std::to_string(static_cast<int>(BE2)) +
+                              " does not support peer access",
+                          __FILE__, __func__, __LINE__);
```

Comment:
`sycl::backend` supports stream insertion overloading, so it can be simplified, like:
```suggestion
            std::ostringstream os;
            os << "Backend " << BE1 << " does not support peer access";
            error_handler(os.str(), __FILE__, __func__, __LINE__);
```
or in similar way

Reply:
While printing integer value does not seem informative about backend name